snmp/tests/common.py

```diff
@@ -212,3 +212,7 @@ def generate_v3_instance_config(metrics, name=None, user=None, auth=None, auth_k
 
 def create_check(instance):
     return SnmpCheck('snmp', {}, [instance])
+
+
+def assert_common_metrics(aggregator, tags=None):
+    aggregator.assert_metric('snmp.devices_monitored', metric_type=aggregator.GAUGE, tags=tags)
```

Comment:
Would there be any value in adding a test specifically to test the behavior of `snmp.devices_monitored`? Eg setup a test with 3 profiles, and make sure that the metric is submitted in exactly 3 copies with the correct tags (in particular `autodiscovery_subnet`, which I don't think we actually test here?).

Reply:
@florimondmanca Made few changes.

- The implementation was not complete: https://github.com/DataDog/integrations-core/pull/6941/commits/e1da3528dfb8ddbe6d9d9de4e054b2715dee6ff4
- Added a test for multiples devices, mocking the cache since it's hard to simulate multiple devices: https://github.com/DataDog/integrations-core/pull/6941/commits/697117abfe12e1447879b2151fabf07b33b6b5f6